webapp/packages/plugin-data-viewer/src/ValuePanelPresentation/ImageValue/ImageValuePresentation.tsx

```diff
@@ -225,7 +224,7 @@ export const ImageValuePresentation: TabContainerPanelComponent<IDataValuePanelP
       return (
         <Container vertical>
           <Container fill overflow center>
-            <QuotaPlaceholder limit={limit} size={valueSize}>
+            <QuotaPlaceholder limit={limitInfo?.limitWithSize}>
```

Comment:
seems like you can pass `elementKey` to the `QuotaPlaceholder` as a property and access `getLimitInfo` from inside the component

(this will reduce logic in the `ImageValuePresentation` component

Reply:
To remove 1 line of a code 
`const limitInfo = data.selectedCell ? data.contentAction.getLimitInfo(data.selectedCell) : null;`

You need to:
1. Pass `elementKey` as prop - you right
2. Pass data action as prop or create your own `contentAction`, which lead to pass 2 more props (`model` and `resultIndex`)

- First option where we pass two props instead of one seems redundant cause we can do this action in component above and there is no need to pass this 2 props to do same action
- Second option: we pass 3 props - logic duplication from component above

I think `QuotaPlaceholder` should be as dumb component as it is. It just draws the layout and thats it. All the business logic supposed to be in a parent component for now

Reply:
With this approach, we can write the whole application in a single component to reduce boilerplate and duplication.

The main goal here is to split logic so we can use this component more widely and modify it without the need to modify every place where we use it